plugins/com.google.cloud.tools.eclipse.appengine.facets.test/src/com/google/cloud/tools/eclipse/appengine/facets/ui/AppEngineStandardProjectConvertCommandHandlerTest.java

```diff
@@ -56,6 +66,22 @@
   @Before
   public void setUp() throws CoreException {
     facetedProject = ProjectFacetsManager.create(projectCreator.getProject());
+
+    // Workaround deadlock bug described in Eclipse bug (https://bugs.eclipse.org/511793).
+    // There are graph update jobs triggered by the completion of the CreateProjectOperation
+    // above (from resource notifications) and from other resource changes from modifying the
+    // project facets. So we force the dependency graph to defer updates.
+    IDependencyGraph.INSTANCE.preUpdate();
+    try {
+      Job.getJobManager().join(DependencyGraphImpl.GRAPH_UPDATE_JOB_FAMILY, null);
+    } catch (OperationCanceledException | InterruptedException ex) {
```

Comment:
What if we don't catch these and let them go through? Personally, I prefer to error out if anything doesn't go intended in tests. (You don't intend to periodically check if this is logged, right?) However, if you believe this kind of exceptions can be ignored, I'm good.